contracts/Rentals.sol

```diff
@@ -563,9 +566,24 @@ contract Rentals is
             )
         );
 
-        address signer = ECDSAUpgradeable.recover(offerHash, _offer.signature);
+        _verifySigner(offerSigner, offerHash, _offer.signature);
+    }
 
-        require(signer == _offer.signer, "Rentals#_verifyOfferSigner: SIGNER_MISMATCH");
+    /// @dev Verify that the signature is valid for the provided signer and hash.
+    /// Will perform an ecrecover for EOA _signers and ERC1271 verification for contract _signers.
+    function _verifySigner(
+        address _signer,
+        bytes32 _hash,
+        bytes memory _signature
+    ) private view {
+        if (_signer.code.length == 0) {
+            require(_signer == ECDSAUpgradeable.recover(_hash, _signature), "Rentals#_verifySigner: SIGNER_MISMATCH");
```

Comment:
Should we need to verify if `signer != 0` as well?

Reply:
when recovering the address from the signature + hash, the OZ library checks that. 
```
if (signer == address(0)) {
    return (address(0), RecoverError.InvalidSignature);
}
```

Reply:
```
function recover(
    bytes32 hash,
    uint8 v,
    bytes32 r,
    bytes32 s
) internal pure returns (address) {
    (address recovered, RecoverError error) = tryRecover(hash, v, r, s);
    _throwError(error);
    return recovered;
}
```